x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinatorTests.java

```diff
@@ -2350,6 +2353,79 @@ public void testFollowerIndexIsCreatedInExecuteAutoFollow() {
         assertThat(autoFollowResults.v2().contains(indexName), equalTo(true));
     }
 
+    public void testRemovesStatsOnRemovingAutoFollowPattern() {
+        // given auto-follow pattern added
+        var autoFollowCoordinator = createAutoFollowCoordinator();
+        autoFollowCoordinator.updateAutoFollowers(
+            createClusterStateWith(
+                createAutoFollowPattern("pattern1", "logs-*"),
+                createAutoFollowPattern("pattern2", "logs-*"),
+                createAutoFollowPattern("pattern3", "logs-*")
+            )
+        );
+
+        // and stats are produced
+        autoFollowCoordinator.updateStats(
+            List.of(
+                new AutoFollowCoordinator.AutoFollowResult("pattern1", new RuntimeException("ClusterStateFetchException")),
+                new AutoFollowCoordinator.AutoFollowResult(
+                    "pattern1",
+                    List.of(Tuple.tuple(new Index("index1", UUIDs.base64UUID()), new RuntimeException("AutoFollowExecutionException")))
+                )
+            )
+        );
+        // and applied
+        assertThat(autoFollowCoordinator.getStats().getRecentAutoFollowErrors(), hasKey("pattern1:index1"));
+
+        // when auto-follow pattern `pattern1` is removed
+        autoFollowCoordinator.updateAutoFollowers(
+            createClusterStateWith(createAutoFollowPattern("pattern2", "logs-*"), createAutoFollowPattern("pattern3", "logs-*"))
+        );
+
+        // then stats are removed as well
+        assertThat(autoFollowCoordinator.getStats().getRecentAutoFollowErrors().keySet(), empty());
+    }
+
+    private AutoFollowCoordinator createAutoFollowCoordinator() {
+        return new AutoFollowCoordinator(
+            Settings.EMPTY,
+            null,
+            mockClusterService(),
+            new CcrLicenseChecker(() -> true, () -> false),
+            () -> 1L,
+            () -> 1L,
+            Runnable::run
+        );
+    }
+
+    private ClusterState createClusterStateWith(AutoFollowPattern... patterns) {
+        var patternsAsMap = Stream.of(patterns).collect(toMap(AutoFollowPattern::getRemoteCluster, Function.identity()));
```

Comment:
Please confirm my understanding: keys in the map are actually RemoteCluster names and this is same as pattern

Reply:
I think we may have touched on this offline, but the keys in the map are actually auto-follow pattern names, which is not the same as the remote cluster name.

Reply:
Correct, I believe this diff is outdated. I will hide it